libraries/provider_firewall_ufw.rb

```diff
@@ -0,0 +1,60 @@
+#
+# Author:: Seth Chisamore (<schisamo@opscode.com>)
+# Cookbook Name:: firewall
+# Resource:: default
+#
+# Copyright:: 2011, Opscode, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+class Chef
+  class Provider::FirewallUfw < Provider
+    include Poise
+    include Chef::Mixin::ShellOut
+
+    def action_enable
+      # prints all the firewall rules
+      # pp @new_resource.subresources
+      if active?
+        Chef::Log.debug("#{@new_resource} already enabled.")
+      else
+        shell_out!('echo yes | ufw enable')
+        Chef::Log.info("#{@new_resource} enabled")
+        if @new_resource.log_level
+          shell_out!("ufw logging #{@new_resource.log_level}")
```

Comment:
Prefer the array form of shell_out `shell_out!('ufw', 'logging', new_resource.log_level)`. Also don't use `@new_resource` directly, the accessor is better because it is more consistent.